app/tutorials/traffic-light-communication/src/main/java/org/eclipse/mosaic/app/tutorial/VehicleToTrafficLightApp.java

```diff
@@ -33,21 +36,38 @@ private void sendGeocastMessage() {
                 .getAdHocModule()
                 .createMessageRouting()
                 .geoBroadCast(geoCircle);
-        getOs().getAdHocModule().sendV2xMessage(new GreenWaveMsg(routing, TrafficLightApp.SECRET));
+        getOs().getAdHocModule().sendV2xMessage(new GreenWaveMsg(routing, TrafficLightApp.SECRET, getOs().getPosition()));
+        getLog().infoSimTime(this, "Sent secret passphrase");
+    }
+
+    //Use TopoBroadcast instead of GeoBroadcast because latter is not compatible with OMNeT++ or ns-3
+    private void sendTopoBroadcastMessage() {
+        int hops = 2;
+        final MessageRouting routing = getOperatingSystem()
+                .getAdHocModule()
+                .createMessageRouting()
+                .topoBroadCast();
+        getOs().getAdHocModule().sendV2xMessage(new GreenWaveMsg(routing, TrafficLightApp.SECRET, getOs().getPosition()));
         getLog().infoSimTime(this, "Sent secret passphrase");
     }
 
     private void sample() {
         getOs().getEventManager().addEvent(
                 getOs().getSimulationTime() + TIME_INTERVAL, this
         );
-        sendGeocastMessage();
+        sendTopoBroadcastMessage();
     }
 
     @Override
     public void onStartup() {
         getLog().infoSimTime(this, "Initialize application");
-        getOs().getAdHocModule().enable();
+        AdHocModuleConfiguration configuration = new AdHocModuleConfiguration()
+                .addRadio()
+                .channel(AdHocChannel.CCH)
+                .power(0.3125)
```

Comment:
do we really want to have such small power value here? this should better be increased to a realistic value and the app checks if the sender is close enough (as you already do)